ui/imports/shared/controls/AmountInput.qml

```diff
@@ -0,0 +1,81 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+Input {
+    id: root
+
+    property int maximumLength: 10
+    property var locale: Qt.locale()
+
+    readonly property alias amount: d.amount
+    readonly property bool valid: validationError.length === 0
+    property bool allowDecimals: true
+
+    validationErrorTopMargin: 8
+    fontPixelSize: 13
+    customHeight: 36
+    placeholderText: "0"
```

Comment:
Now, this is a real nitpick, but:

https://doc.qt.io/qt-5/qml-qtqml-locale.html#zeroDigit-prop